### PR TITLE
Refactor Statistics Class

### PR DESCRIPTION
**Pull request by thisisagithubaccounta**
- Refactor Statistics Class

- Add slight error handling

- Add tests

I've added some tests in for the Statistics class. Personally I think it makes development a bit easier and as far as i know they wont be compiled into any artefacts so shouldn't affect the size on maven.

Open to any feedback or suggestions 😃

### CODE REVIEW

File: src/main/java/scientific/Statistics.java [hunk 1]
```diff
@@ -8,132 +8,102 @@
...
+	private static void checkInputHasElements(double[] inputArray) {
+		if(inputArray.length == 0) {
+			throw new IllegalArgumentException("The array of values must contain at least one element");
 		}
+	}
 
-		return total/data.length;
+	public static double mean(double[] valuesToCalculateMeanOf){
+		checkInputHasElements(valuesToCalculateMeanOf);
+		double total = Arrays.stream(valuesToCalculateMeanOf).sum();
```
Comment (thisisagithubaccounta) [inline src/main/java/scientific/Statistics.java:23]: There's also an option to calculate the average using streams too. The only issue with this is that it returns an OptionalDouble which you then have to check for null or not. It makes the function a bit longer but may be more reliable.
```
OptionalDouble average = Arrays.stream(valuesToCalculateMeanOf).average();
if(average.isPresent) {
    return average.getAsDouble;
}
throw new IllegalStateException("Unable to calculate average, OptionalDouble null")
````

File: pom.xml [hunk 2]
```diff
@@ -5,7 +5,19 @@
   <groupId>scientific-java</groupId>
   <artifactId>ScientificJavaLib</artifactId>
   <version>0.0.1-SNAPSHOT</version>
-  <packaging>jar</packaging>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
```
Comment (bowebgit) [inline pom.xml]: Need to specify the version here.

File: pom.xml [hunk 3]
```diff
@@ -5,7 +5,19 @@
   <groupId>scientific-java</groupId>
   <artifactId>ScientificJavaLib</artifactId>
   <version>0.0.1-SNAPSHOT</version>
-  <packaging>jar</packaging>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
```
Comment (blackmathx) [inline pom.xml:12]: Need to specify the version here.

File: pom.xml [hunk 4]
```diff
@@ -15,17 +27,11 @@
...
-	    <version>4.12</version>
-	    <scope>test</scope>
-	</dependency>
--->
-
-
+      <dependency>
+          <groupId>junit</groupId>
+          <artifactId>junit</artifactId>
+          <version>RELEASE</version>
```
Comment (blackmathx) [inline pom.xml]: Specify the junit version here or set to use the latest release following this stackoverflow answer https://stackoverflow.com/questions/30571/how-do-i-tell-maven-to-use-the-latest-version-of-a-dependency

File: pom.xml [hunk 5]
```diff
@@ -5,7 +5,19 @@
...
   <artifactId>ScientificJavaLib</artifactId>
   <version>0.0.1-SNAPSHOT</version>
-  <packaging>jar</packaging>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>8</source>
```
Comment (blackmathx) [inline pom.xml]: I think you want source and target to be 1.8

File: src/main/java/scientific/Statistics.java [hunk 6]
```diff
@@ -8,132 +8,102 @@
 
 package scientific;
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.HashMap;
-import java.util.Map;
+import java.util.*;
```
Comment (bowebgit) [inline src/main/java/scientific/Statistics.java]: Don't import all. Only use what you need. It's important.

File: src/main/java/scientific/Statistics.java (hunk 1 above)
Comment (blackmathx) [reply src/main/java/scientific/Statistics.java:23]: This is fine.

**Comment by blackmathx** [comment]
The variable names you've added in Statistics class are nice to read. Fix the comments in the pom and I'll merge the pull request. Nice work.

File: pom.xml (hunk 5 above)
Comment (thisisagithubaccounta) [reply pom.xml]: Good spots. Should be sorted now.

**Comment by blackmathx** [comment]
Merged.

**Pull request merged**